context-propagation/src/main/java/io/micrometer/context/ContextRegistry.java

```diff
@@ -100,8 +102,8 @@ public <V> ContextRegistry registerThreadLocalAccessor(String key, ThreadLocal<V
      * @return the same registry instance
      * @param <V> the type of value stored in the ThreadLocal
      */
-    public <V> ContextRegistry registerThreadLocalAccessor(String key, Supplier<V> getSupplier, Consumer<V> setConsumer,
-            Runnable resetTask) {
+    public <V extends @Nullable Object> ContextRegistry registerThreadLocalAccessor(String key, Supplier<V> getSupplier,
```

Comment:
This means that `V` can be "nullable" which lets both the supplier return null and the consumer accept null OR it can be non-nullable which means both the supplier and the consumer must return non-null. Same for `getValue` and `setValue`.